### PR TITLE
Update syntax for contract methods

### PR DESCRIPTION
**Pull request by mohammadfawaz**

### CODE REVIEW

File: contracts/Forc.toml [hunk 1]
```diff
@@ -5,5 +5,5 @@ license = "Apache-2.0"
 name = "swayswap"
 
 [dependencies]
-core = { git = "http://github.com/FuelLabs/sway-lib-core" }
-std = { git = "http://github.com/FuelLabs/sway-lib-std" }
+core = { git = "http://github.com/FuelLabs/sway-lib-core", branch = "master" }
+std = { git = "http://github.com/FuelLabs/sway-lib-std", branch = "master" }
```
Comment (adlerjohn) [inline contracts/Forc.toml:9]:
```suggestion
std = { git = "https://github.com/FuelLabs/sway-lib-std", branch = "master" }
```

File: contracts/src/main.sw [hunk 2]
```diff
@@ -161,20 +122,20 @@ fn get_output_price(output_amount: u64, input_reserve: u64, output_reserve: u64)
...
+        let key = key_deposits(sender, (msg_asset_id()).into());
 
         let total_amount = get::<u64>(key) + msg_amount();
         store(key, total_amount);
     }
 
-    fn withdraw(gas_: u64, amount_: u64, asset_id_: b256, params: WithdrawParams) {
-        let asset_id = params.asset_id.into();
+    fn withdraw(amount: u64, asset_id: ContractId) {
+        let asset_id = asset_id.into();
```
Comment (adlerjohn) [inline contracts/src/main.sw:138]: Maybe don't shadow here, and just use `asset_id.into()` as needed?

File: contracts/Forc.toml [hunk 3]
```diff
@@ -5,5 +5,4 @@ license = "Apache-2.0"
 name = "swayswap"
 
 [dependencies]
-core = { git = "http://github.com/FuelLabs/sway-lib-core" }
-std = { git = "http://github.com/FuelLabs/sway-lib-std" }
+std = { git = "http://github.com/FuelLabs/sway-lib-std", branch = "master" }
```
Comment (adlerjohn) [inline contracts/Forc.toml:8]:
```suggestion
std = { git = "https://github.com/FuelLabs/sway-lib-std", branch = "master" }
```

File: contracts/src/main.sw [hunk 4]
```diff
@@ -278,62 +238,62 @@ impl Exchange for Contract {
...
+        assert(msg_amount() > 0 && min > 0);
+        assert((msg_asset_id()).into() == ETH_ID || (msg_asset_id()).into() == TOKEN_ID);
 
         // TODO replace
         // let sender = msg_sender().unwrap();
         let sender = ~Address::from(0x0000000000000000000000000000000000000000000000000000000000000000);
 
-        let eth_reserve = this_balance(ETH_ID);
-        let token_reserve = this_balance(TOKEN_ID);
+        let eth_reserve = this_balance(~ContractId::from(ETH_ID));
```
Comment (adlerjohn) [inline contracts/src/main.sw:250]: Can you file an issue to make these constants `ContractId`s (unless there's already one)?

**Review by adlerjohn — APPROVED** [review]

File: contracts/src/main.sw (hunk 4 above)
Comment (mohammadfawaz) [reply contracts/src/main.sw:250]: You mean something like `const ETH_ID = ~ContractId::from(0x00..00)`? Yeah I think this requires compiler support as we only expect literals on the RHS currently. Let me file an issue.

Comment (adlerjohn) [reply contracts/src/main.sw:250]: Yes

**Pull request merged**